src/prompt_file.py

```diff
@@ -1,262 +1,164 @@
 import os
+from pyexpat import model
 import time
 import configparser
+import pickle
 
 from pathlib import Path
-
+from openai import Model
+from prompt_engine.code_engine import CodeEngine, ModelConfig
 
 API_KEYS_LOCATION = os.path.join(os.path.dirname(os.path.realpath(__file__)), 'openaiapirc')
 
-class PromptFile:
-    context_source_filename = ""
-    default_context_filename = "current_context.txt"
+
+class CodexCLIConfig(ModelConfig):
+    """
+    Interaction class is used to store the model config to be used in the prompt engine
+    """
+    def __init__(self, **kwargs):
+        self.engine = kwargs['engine'] if ('engine' in kwargs and kwargs['engine']) else 'code-davinci-002'
```

Comment:
is `code-davinci-002` globally available and public? I think there was some discussion about helping users detecting available models, e.g. scanning OpenAI API

Reply:
for instance, we could opt for not having a default, requiring an explicit configuration - wdyt?

Reply:
code-davinci-002 is globally available. We can definitely make another command to list the available models to the user. I think it has been implemented in some other repo, but was not present in codex-cli. I think it should come in a separate PR though